### PR TITLE
fix(html): HTML pipeline should use downloader

### LINKED ISSUE #138: HTML pipeline should use downloader
As `helix-pipeline` is in its dependencies, it can use [Downloader.js](https://github.com/adobe/helix-pipeline/blob/master/src/utils/Downloader.js)

### PR DESCRIPTION
**Pull request by dominique-pfister**
fix #138

### CODE REVIEW

File: src/html_json.js [hunk 1]
```diff
@@ -181,12 +157,34 @@ module.exports.main = async (context, action) => {
...
+      return {
+        indices: [],
+      };
+    }
+    return (await new IndexConfig()
+      .withSource(indexYAML.body)
+      .init()).toJSON();
+  };
+
+  const config = await loadConfig();
```
Comment (dominique-pfister) [inline src/html_json.js]: Using a function *inside* a function, to be able to use the arguments in the outer scope, not sure, whether this is discouraged, though.

File: src/html_json.js [hunk 2]
```diff
@@ -181,12 +157,34 @@ module.exports.main = async (context, action) => {
   const {
     owner, repo, ref, path,
   } = action.request.params;
+
   const { logger } = action;
 
+  const secrets = setdefault(action, 'secrets', {});
+  secrets.REPO_RAW_ROOT = secrets.REPO_RAW_ROOT || 'https://raw.githubusercontent.com/';
+  secrets.HTTP_TIMEOUT = secrets.HTTP_TIMEOUT || 10000;
```
Comment (dominique-pfister) [inline src/html_json.js]: Use a default value for `secrets`, not sure, whether there ever is the need for that.

File: package.json [hunk 3]
```diff
@@ -35,6 +35,8 @@
   "dependencies": {
     "@adobe/helix-cli": "8.1.6",
     "@adobe/helix-pipeline": "6.9.1",
+    "@adobe/helix-shared": "^6.0.4",
```
Comment (tripodsan) [inline package.json]:
```suggestion
    "@adobe/helix-shared": "6.0.4",
```

File: package.json [hunk 4]
```diff
@@ -35,6 +35,8 @@
...
+    "ferrum": "^1.7.0",
```
Comment (tripodsan) [inline package.json]:
```suggestion
    "ferrum": "1.7.0",
```

File: src/html_json.js [hunk 5]
```diff
@@ -181,12 +157,34 @@ module.exports.main = async (context, action) => {
...
+
+  const loadConfig = async () => {
+    const downloader = new Downloader(context, action, {
```
Comment (tripodsan) [inline src/html_json.js]: why not use `action.downloader` ?

File: src/html_json.js (hunk 1 above)
Comment (tripodsan) [reply src/html_json.js]: I think this is ok.

File: test/html_json.test.js [hunk 6]
```diff
@@ -39,10 +39,6 @@ describe('HTML Indexing', () => {
 
   before(async () => {
     testRoot = await createTestRoot();
-    nock('https://raw.githubusercontent.com')
-      .get(/\/helix\/github.com--adobe--helix-index-pipelines\/.*\/helix-query.yaml/)
-      .replyWithFile(200, resolve(specsDir, 'blog', 'helix-query.yaml'))
-      .persist();
     nock('https://github.com--adobe--helix-index-pipelines-helix.project-helix.page')
       .get((uri) => uri.startsWith('//test/specs/blog'))
```
Comment (tripodsan) [inline test/html_json.test.js]: why double slash? probably something wrong with the URL

File: src/html_json.js (hunk 5 above)
Comment (dominique-pfister) [reply src/html_json.js]: Didn't know, but what about `forceHttp1`, can I somehow pass that option?

File: test/html_json.test.js (hunk 6 above)
Comment (dominique-pfister) [reply test/html_json.test.js]: Yep, the `fetch` property in `helix-query.yaml` was malformed.

File: src/html_json.js (hunk 5 above)
Comment (tripodsan) [reply src/html_json.js]: for the tests, use `process.env.HELIX_PIPELINE_FORCE_HTTP1`

see: https://github.com/adobe/helix-pipeline/blob/ec0995b82728d5956dbdb551d1089c7938c9b561/src/utils/Downloader.js#L36

File: src/html_json.js [hunk 7]
```diff
@@ -165,11 +164,7 @@ module.exports.main = async (context, action) => {
   secrets.HTTP_TIMEOUT = secrets.HTTP_TIMEOUT || 10000;
```
Comment (tripodsan) [inline src/html_json.js]: I think, now you also don't need the secrets anymore....

**Pull request merged**